freshmaker/image_verifier.py

```diff
@@ -22,120 +22,119 @@
         """
-        self.lb = lb if lb else LightBlue(
-            server_url=conf.lightblue_server_url,
-            cert=conf.lightblue_certificate,
-            private_key=conf.lightblue_private_key)
+        self.pgql = (
+            pgql
+            if pgql
+            else PyxisGQL(
+                url=conf.pyxis_server_url,
+                cert=(
+                    conf.pyxis_certificate,
+                    conf.pyxis_private_key,
+                ),
+            )
+        )
 
     def _verify_repository_data(self, repo):
         """
-        Verifies the Lightblue ContainerRepository data.
+        Verifies the PyxisGQL ContainerRepository data.
         Raises ValueError in case of error.
         """
-        categories = set(conf.lightblue_release_categories)
+        categories = set(conf.container_release_categories)
         repo_categories = repo.get("release_categories", [])
+
         if not set(repo_categories).intersection(categories):
             raise ValueError(
                 "Only repositories with one of %r release_categories can be "
-                "rebuilt, but found %r." % (categories, repo_categories))
+                "rebuilt, but found %r." % (categories, repo_categories)
+            )
 
         if not repo["published"]:
             for exc in conf.unpublished_exceptions:
-                if repo["registry"] == exc["registry"] and repo["repository"] == exc["repository"]:
+                if repo["repository"] == exc["repository"]:
                     break
             else:
                 raise ValueError(
                     "Only published repositories or unpublished exceptions can be rebuilt, but "
                     "this repository is not published.")
 
-        if "auto_rebuild_tags" not in repo:
-            raise ValueError(
-                "The \"auto_rebuild_tags\" in COMET is not set.")
+        if "auto_rebuild_tags" not in repo or repo["auto_rebuild_tags"] is None:
+            raise ValueError('The "auto_rebuild_tags" in COMET is not set.')
 
         if "auto_rebuild_tags" in repo and repo["auto_rebuild_tags"] == []:
             raise ValueError(
-                "The \"auto_rebuild_tags\" in COMET is set to an empty list, this means "
-                "rebuilds of images in this repository are disabled.")
+                'The "auto_rebuild_tags" in COMET is set to an empty list, this means '
+                "rebuilds of images in this repository are disabled."
+            )
 
     def _verify_image_data(self, image):
         """
-        Verifies the Lightblue ContainerImage data.
+        Verifies the PyxisGQL ContainerImage data.
         Raises ValueError in case of error.
         """
+        if "brew" not in image:
+            raise ValueError("'brew' field not set in image")
+
+        if "build" not in image["brew"]:
+            raise ValueError("'brew.build' field not set in image")
+
         if not image["content_sets"]:
             raise ValueError(
```

Comment:
While testing `verify_image` and `verify_repository`, sometimes it was failing on this exception because an image had empty content_sets, but I suppose that's alright because the methods were throwing exceptions when they were supposed to?

Reply:
Yes, it's supposed to be, when image is missing `content_sets`, freshmaker is unable to rebuild it, and the API should report error.